engine/gameobject/src/gameobject/gameobject.h

```diff
@@ -391,6 +391,9 @@ namespace dmGameObject
      * tracked by the collection, e.g resource.release(id) has been called
      */
     void RemoveDynamicResourceHash(HCollection collection, dmhash_t resource_hash);
+
+    // For tests
+    void ResetInstanceIndex();
```

Comment:
I don't like it. Should I rewrite these tests to make sure we don't expect exact names?

Reply:
Sure, it may be beneficial to not rely on the names?
And also one function less to take into consideration when testing.

Reply:
Fixed!